src/views/TabView/TabBarBottom.js

```diff
@@ -187,7 +193,7 @@ class TabBarBottom extends React.PureComponent<Props> {
           {routes.map((route: NavigationRoute, index: number) => {
             const focused = index === navigation.state.index;
             const scene = { route, index, focused };
-            const onPress = getOnPress(scene);
+            const onPress = getOnPress(previousScene, scene);
```

Comment:
As noted by @spencercarli this is a breaking change. I think we shouldn't introduce breaking changes unless there are very good reasons for doing so. In this case, changing `const onPress = getOnPress(previousScene, scene);` to `const onPress = getOnPress(scene, previousScene);` and changes in a couple more lines will make this a non-breaking change. Imho it's better to do the change here rather to ask users to do it.

It is truth that the order of params in `onPress(previousScene, scene, jumpToIndex)  is slightly more intuitive than `onPress(scene, jumpToIndex, previousScene)`, but I don't think this is a reason for making a breaking change. If there is no other reason for doing the breaking change, I'd say we don't do it. Any thoughts?

Reply:
or perhaps a compromise (still breaking, but less), if you want to have a nicer order of params.

change
```
  getOnPress: (
    previousScene: NavigationRoute,
    scene: TabScene
  ) => (
    previousScene: NavigationRoute,
    scene: TabScene,
    jumpToIndex: (index: number) => void
  ) => void,
```

to

```
  getOnPress: (
    scene: TabScene.
    previousScene: NavigationRoute,
  ) => (
    scene: TabScene,
    previousScene: NavigationRoute,
    jumpToIndex: (index: number) => void
  ) => void,
`

Reply:
I see your point, and to be honest I'm not sure what could be a sensible decision: generally, I prefer a meaningful order for function arguments but the breaking change is not a small one.

Given that this is still a _beta_ version, I'd be ready as a developer to accept breaking changes if needed.

Reply:
I'm back and forth on this API change. I agree with @pietro909 in that this is beta software, therefore breaking changes will happen. This is also a relatively new feature of the library so I'm not sure how many people are actively using this.

On the flip side of that, we want to encourage confidence and minimize breaking changes and @vonovak's first solution would allow that.

Another question: If we do make a breaking change now, what can we do to ensure we don't have another breaking change in the future? Would it be to pass an object with the arguments in there (eliminates the order issue)?

Reply:
I find the idea of an object more scalable as a pattern, like:

    ({
        scene: TabScene,
        previousScene: NavigationRoute,
        jumpToIndex: (index: number)
    }) => void

This one offers two advantages:
1. order doesn't matter anymore
2. can add parameters without breaking the API

On the other hand, it makes less clear what is being passed if the consumer is not using Flow (or Typescript). But I think that this is the reason why we have docs :-)

Reply:
Exactly - docs are there and we can give added flexibility in the future. @vonovak @kelset any hesitations on that API change? Still breaking but "better" breaking? 😄

Reply:
⛵️

Reply:
Do we have any news? Can I proceed with the object instead of single arguments?

Reply:
Let's ping @vonovak ;) Btw I think you still need to remove the comment at line 140 commented by him a while back

Reply:
Yes, I was also thinking about that (using a single object argument), I think its okay to do it that way, so let's go ahead with that.